### DISCUSSION TITLE
Some ideas for small usability improvements

### DISCUSSION THREAD

**Discussion by FlooferLand** [Leafwing-Studios/leafwing-input-manager#631] [Ideas]
Hello, its me again! _(managed to get the latest `actionlike` version working via git)_
Here are some things I'd really appreciate that wouldn't be too trivial to implement:

1. Adding optional tuple support to `InputMap.insert`
   * Bevy does "functions that can take in a singular value _or_ a tuple" a lot
   * It would let you do things like this _(I originally tried this but it didn't work)_
     ```rs
     input_map.insert(GamePlayerAction::Crouch, (KeyCode::ControlLeft, KeyCode::ControlRight));
     ```
   * `insert_dual_axis` could benefit from this, it would be cleaner/simpler if you were able to do the following to add in a virtual mapping:
     ```rs
     input_map.insert_dual_axis(GamePlayerAction::Move, (KeyCode::A, KeyCode::D));
     ```
2. `MouseVirtualAxis` _(and `OtherVirtualAxis`?)_
    * In some cases people might want to map the mouse buttons to an axis _(I originally tried this while setting up zooming using the mouse wheel, though `MouseScrollAxis` does exist)_. 
    * _PS: I think combining the `(Mouse/Keyboard/Gamepad)VirtualAxis` type into a general `VirtualAxis` type would be neater. Same with the `*Dpad/*Dpad3d` types_
3. Perhaps more features/examples for player-bindable controls?

**Comment by alice-i-cecile** [comment]
1. Would be nice, but will hurt compile times more than is really worth it. Once Rust has real variadics support I would love to add this.
2. I'm on board with this; please submit a PR or dedicated issue.
3. Totally agree: this is the last key workflow that we really need to polish and teach.

**Reply by FlooferLand** [reply]
Made one at #632!
Originally tried making a PR but I couldn't get my head around why it was telling me that Bevy's KeyCode doesn't implement `Buttonlike` _(even though it does!)_
